src/config.cpp

```diff
@@ -716,46 +740,57 @@ CircuitConfig::CircuitConfig(const std::string& contents, const std::string& bas
     Parser parser(contents, basePath);
 
     _expandedJSON = parser.getExpandedJSON();
-
-    _components = parser.parseDefaultComponents();
+    _circuitType = parser.getCircuitType();
 
     _nodeSetsFile = parser.getNodeSetsPath();
 
     // Load node population overrides and check biophysical types
-    _nodePopulationProperties = parser.parseNodePopulations();
+    _nodePopulationProperties = parser.parseNodePopulations(_circuitType);
 
     // Load edge population overrides
-    _edgePopulationProperties = parser.parseEdgePopulations();
+    _edgePopulationProperties = parser.parseEdgePopulations(_circuitType);
+
+    Components components = parser.parseDefaultComponents();
 
     const auto updateDefaultProperties =
         [&](std::unordered_map<std::string, PopulationProperties>& map,
             const std::string& defaultType) {
             for (auto& entry : map) {
-                if (entry.second.type.empty()) {
-                    entry.second.type = defaultType;
+                auto& component = entry.second;
+                if (component.type.empty()) {
+                    component.type = defaultType;
                 }
-                if (entry.second.alternateMorphologyFormats.empty()) {
-                    entry.second.alternateMorphologyFormats = _components.alternateMorphologiesDir;
+
+                if (component.alternateMorphologyFormats.empty()) {
+                    component.alternateMorphologyFormats = components.alternateMorphologiesDir;
                 }
-                if (entry.second.biophysicalNeuronModelsDir.empty()) {
-                    entry.second.biophysicalNeuronModelsDir =
-                        _components.biophysicalNeuronModelsDir;
+
+                if (component.biophysicalNeuronModelsDir.empty()) {
+                    component.biophysicalNeuronModelsDir = components.biophysicalNeuronModelsDir;
                 }
-                if (entry.second.morphologiesDir.empty()) {
-                    entry.second.morphologiesDir = _components.morphologiesDir;
+
+                if (component.morphologiesDir.empty()) {
+                    component.morphologiesDir = components.morphologiesDir;
                 }
             }
         };
+
     updateDefaultProperties(_nodePopulationProperties, "biophysical");
     updateDefaultProperties(_edgePopulationProperties, "chemical_synapse");
 
-    checkBiophysicalPopulations(_nodePopulationProperties);
+    if (_circuitType == Type::Complete) {
+        checkBiophysicalPopulations(_nodePopulationProperties);
```

Comment:
I can't spot any side effects in `checkBiophysicalPopulations`. Therefore, doing this conditionally should be correct.

Reply:
yeah, perhaps it should be name `checkInvariantsBiophysicalPopulations` or `throwIfNonconformingPopulations`?

Reply:
Not on my account. I'm only saying that `checkBiophysicalPopulations` might have done something like also set a default for example, but I looked and that's not the case, as far as I can tell.